### PR TITLE
Adjust env markers and deps for Python 3.8+

### PR DESCRIPTION
**Pull request by bdraco**
followup to #997

https://github.com/aio-libs/multidict/commit/4fffc3d5d647d631f8830282d16ba7260dabea39#r146458357
https://github.com/aio-libs/multidict/commit/4fffc3d5d647d631f8830282d16ba7260dabea39#r146458343

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -155,8 +155,7 @@ repos:
     additional_dependencies:
     - types-docutils
     - lxml  # dep of `--txt-report`, `--cobertura-xml-report` & `--html-report`
-    - pytest < 8
```
Comment (webknjaz) [inline .pre-commit-config.yaml:158]: I think `pytest` is still needed, just not capped. MyPy is linting the tests too.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -155,8 +155,7 @@ repos:
...
-    - Sphinx >= 5.3.0, < 6
+    - Sphinx >= 5.3.0
```
Comment (webknjaz) [inline .pre-commit-config.yaml:158]:
```suggestion
    - pytest
    - Sphinx >= 5.3.0
```

File: .pre-commit-config.yaml (hunk 1 above)
Comment (bdraco) [reply .pre-commit-config.yaml:158]: Yeah that makes more sense.... getting tired 💤

**Review by webknjaz — APPROVED** [review]

**Comment by webknjaz** [comment]
@bdraco missed removing the conditional imports: https://app.codecov.io/github/aio-libs/multidict/commit/6197e510d9c11814524f1a48117cf2a45ba1a59a/blob/tests/conftest.py?flags%5B0%5D=pytest#L11.

**Comment by bdraco** [comment]
I checked the main code base but missed the tests!  I’ll take a look when I get back back from finding dinner before I go to bed

**Pull request merged**